react/features/tensorflow-effects/JitsiStreamCropPersonEffect.js

```diff
@@ -0,0 +1,198 @@
+        videoDiv.style.display = 'none';
+
+        this._maskCanvasElement.width = this._cameraWidth;
+        this._maskCanvasElement.height = this._cameraHeight;
+        this._maskCanvasContext = this._maskCanvasElement.getContext('2d');
+        this._outputCanvasElement.width = this._width;
+        this._outputCanvasElement.height = this._height;
+
+        this._videoFrameTimerWorker.postMessage({
+            id: SET_INTERVAL,
+            timeMs: 1000 / this._cameraFrameRate
+        });
+        this._maskFrameTimerWorker.postMessage({
+            id: SET_INTERVAL,
+            timeMs: 50
+        });
+
+        return this._outputCanvasElement.captureStream(this._cameraFrameRate);
+    }
+
+    /**
+     * Stops the capture and render loop.
+     *
+     * @returns {void}
+     */
+    stopEffect() {
+        this._videoFrameTimerWorker.postMessage({
+            id: CLEAR_INTERVAL
+        });
+        this._maskFrameTimerWorker.postMessage({
+            id: CLEAR_INTERVAL
+        });
+    }
+
+    /**
+     * Loop function to render the canvas and draw the crop effect.
+     *
+     * @private
+     * @returns {void}
+     */
+    _renderVideo() {
+        this._maskCanvasContext.drawImage(this._inputVideoElement, 0, 0,
+            this._cameraWidth, this._cameraHeight);
+        if (this._segmentationData) {
+            const img = this._maskCanvasContext.getImageData(0, 0,
+                this._cameraWidth, this._cameraHeight);
+            const pixels = img.data;
+            const data = this._segmentationData.data;
+
+            for (let i = 0; i < data.length; i++) {
+                // if the pixel doesn't belong to the person, set the
+                // alpha channel value to 0 to make it transparent.
+                pixels[(i * 4) + 3] = data[i] === 0 ? 0 : 255;
+            }
+            this._maskCanvasContext.putImageData(img, 0, 0);
+        }
+        this._ctx.drawImage(this._inputDesktopElement, 0, 0, this._width,
+            this._height);
+        this._ctx.drawImage(this._maskCanvasElement, this._width - this._cameraWidth,
+            this._height - this._cameraHeight, this._cameraWidth, this._cameraHeight);
+    }
+
+    /**
+     * Loop function to render the background mask.
+     *
+     * @private
+     * @returns {void}
+     */
+    _renderMask() {
+        this._bpModel.estimatePersonSegmentation(
+            this._inputVideoElement,
+            8, // Chose 32 for better performance
+            0.55 // Represents probability that a pixel belongs to a person
+        )
+        .then(value => {
+            this._segmentationData = value;
+        });
+    }
+
+    /**
+     * Checks if the local track supports this effect.
+     *
+     * @param {JitsiLocalTrack} jitsiLocalTrack - Track to apply effect.
+     * @returns {boolean} - Returns true if this effect can run on the specified track
+     * false otherwise.
+     */
+    isEnabled(jitsiLocalTrack) {
```

Comment:
We would need a change in LJM as well if I change it here. We call isEnabled on the effect before applying it. This would also mean we have to change it for JitsiStreamBlurEffect as well.